sdk/resourcemanager/Azure.ResourceManager/src/autorest.md

```diff
@@ -593,6 +589,73 @@ directive:
+  # this gives very little information to our generator and make it to behave weirdly
+  # this directive changes its path to a normal scope resource path
+  - from: links.json
+    where: $.paths
+    transform: >
+      var oldLinkResource = $["/{linkId}"];
+      delete $["/{linkId}"];
+      $["/{scope}/providers/Microsoft.Resources/links/{name}"] = oldLinkResource;
+      for (var r in oldLinkResource) {
+          var things = oldLinkResource[r];
+          var parameters = things.parameters;
+          // remove the parameter with the name of "linkId", and add two new parameters "scope" and "name"
+          if (r == "put") {
+            things.parameters = [
+                {
+                    "name": "scope",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "The fully qualified ID of the resource link. Use the format, /subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/{provider-namespace}/{resource-type}/{resource-name}/Microsoft.Resources/links/{link-name}. For example, /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myGroup/Microsoft.Web/sites/mySite/Microsoft.Resources/links/myLink",
+                    "x-ms-skip-url-encoding": true
+                },
+                {
+                    "name": "name",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "fuck you"
+                },
+                {
+                    "name": "parameters",
+                    "in": "body",
+                    "required": true,
+                    "schema": {
+                    "$ref": "#/definitions/ResourceLink"
+                    },
+                    "description": "Parameters for creating or updating a resource link."
+                },
+                {
+                    "$ref": "#/parameters/ApiVersionParameter"
+                }
+            ]
+          } else {
+            things.parameters = [
+                {
+                    "name": "scope",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "The fully qualified ID of the resource link. Use the format, /subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/{provider-namespace}/{resource-type}/{resource-name}/Microsoft.Resources/links/{link-name}. For example, /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myGroup/Microsoft.Web/sites/mySite/Microsoft.Resources/links/myLink",
+                    "x-ms-skip-url-encoding": true
+                },
+                {
+                    "name": "name",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "fuck you"
```

Comment:
（￣.￣）Remember to change this description

Reply:
Oops I forget to change.... Thank you to find out